### PR TITLE
native version function diag

### PR DESCRIPTION
**Pull request by Rubtsowa**

### CODE REVIEW

**Review by samir-nasibli — COMMENTED** [review]
Provide numpy original tests that checks you feature.
No tests for corner cases.

File: dpnp/dpnp_algo/dpnp_algo_arraycreation.pyx [hunk 1]
```diff
@@ -54,38 +54,36 @@ __all__ += [
...
-    elif v.ndim == 2:
-        # computation of result size
-        size0 = min(v.shape[0], v.shape[0] + k)
-        size1 = min(v.shape[1], v.shape[1] - k)
-        size = min(size0, size1)
+        shape_result = (size, size)
+    else:
+        size = min(v.shape[0], v.shape[0] + k, v.shape[1], v.shape[1] - k)
         if size < 0:
             size = 0
```
[the remark this answers is not in the document]
Comment (Rubtsowa) [reply dpnp/dpnp_algo/dpnp_algo_arraycreation.pyx:72]: @samir-nasibli  looks this https://github.com/IntelPython/dpnp/pull/621/files#diff-d7986221bd1f8eb7a9e1c069323072d09223f2929098ef920f4ed0ba596e391aR70

we have case when, for example, k = -10 and v.shape = (2, 3), then 
size = min(2, 2-10, 3, 3+10)
size = min(2, -8, 3, 13)
=> size == -8 

this out our array, then we need empty array, i.e. size = 0.

[the remark this answers is not in the document]
Comment (samir-nasibli) [reply dpnp/dpnp_algo/dpnp_algo_arraycreation.pyx:72]: it looks like the chosen variable name (`size`) is not very good.
Logically variable with name `size` should be non-negative value, it might be a little bit confusing.
I propose to use just `n1` or `n_`.

**Review by shssf — APPROVED** [review]
